authoringAssistant/src/main/java/explorableviz/transparenttext/variable/ValueOptions.java

```diff
@@ -0,0 +1,138 @@
+    }
+    public abstract Variables expandVariable(Random random, String varName);
+
+    public static class StringValue extends ValueOptions {
+        private final String value;
+
+        public StringValue(String value) {
+            this.value = value;
+        }
+
+        @Override
+        public String get() {
+            return value;
+        }
+
+        @Override
+        public Variables expandVariable(Random random, String varName) {
+            Variables v = new Variables();
+            v.put(varName, switch (value) {
+                case "RANDOM_INT" -> new ValueOptions.Number(random.nextInt(10));
+                case "RANDOM_FLOAT" -> new ValueOptions.Number(random.nextFloat() * 10);
+                case "RANDOM_STRING" -> new ValueOptions.StringValue(getRandomString(8, random).toLowerCase());
+                default -> new StringValue(value);
+            });
+            return v;
+        }
+    }
+
+    public static class Number extends ValueOptions {
+        private final BigDecimal value;
+
+        public Number(float value) {
+            this.value = BigDecimal.valueOf(value);
+        }
+
+        @Override
+        public BigDecimal get() {
+            return value;
+        }
+
+        @Override
+        public Variables expandVariable(Random random, String varName) {
+            Variables v = new Variables();
+            v.put(varName, this);
+            return v;
+        }
+    }
+
+    public static class Map extends ValueOptions {
+        private final HashMap<String, String> value;
+
+        public Map(HashMap<String, String> value) {
+            this.value = value;
+        }
+
+        public void add(String key, String value) {
+            this.value.put(key, value);
+        }
+
+        @Override
+        public HashMap<String, String> get() {
+            return value;
+        }
+
+        @Override
+        public Variables expandVariable(Random random, String varName) {
+            Variables v = new Variables();
+            keySet().forEach(k -> v.put(STR."\{varName}.\{k}", new StringValue(value.get(k))));
+            return v;
+        }
+
+        public String getValue(String key) {
+            return value.get(key);
+        }
+
+        public Set<String> keySet() {
+            return this.value.keySet();
+        }
+    }
+
+    public static class List extends ValueOptions {
+        private final java.util.List<ValueOptions> value;
+
+        public List(java.util.List<ValueOptions> value) {
+            this.value = value;
+        }
+
+        @Override
+        public java.util.List<ValueOptions> get() {
+            return value;
+        }
+
+        @Override
+        public Variables expandVariable(Random random, String varName) {
+            return value.get(random.nextInt(value.size())).expandVariable(random, varName);
+        }
+    }
+
+    private static String getRandomString(int length, Random generator) {
```

Comment:
Maybe create a new issue for the variable naming thing we discussed earlier (replacing random variable names by more carefully chosen ones, e.g. meaningful ones, meaningless ones, meaningful but misleading ones).

Reply:
#87